### PR TITLE
Implementing search method

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing search method

### CODE REVIEW

File: TheMET/TheMET/Network/MetAPI.swift [hunk 1]
```diff
@@ -73,5 +73,74 @@ class MetAPI{
...
+            }
+    }
+    
+    private func fillingParameters (parameters: [SearchParameter]) -> [String : String] {
+        var searchParameters: [String : String] = [ : ]
+        for parameter in parameters {
+            switch parameter {
+            case .q(let text):
+                searchParameters["q"] = text
+            case .isHighlight(let bool):
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/MetAPI.swift:106]: `bool`: bad name
please provide meaningful name for the associated values.

File: TheMET/TheMET/Network/MetAPI.swift [hunk 2]
```diff
@@ -73,5 +73,74 @@ class MetAPI{
...
+        if value {
+            return ("true")
+        } else {
+            return ("false")
+        }
+    }
+    
+    private func covertDatasToString(from: Int, to: Int) -> String {
+        var result: String
+        result = String(describing: from)
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/MetAPI.swift:141]: `String(describing:)` is too complicated way. Usually it is used on objects/structs. 
`String(12)` works much faster than `String(describing: 12)`

File: TheMET/TheMET/Network/MetAPI.swift [hunk 3]
```diff
@@ -73,5 +73,74 @@ class MetAPI{
...
+                    } catch {
+                        completion(nil)
+                    }
+                } else {
+                    completion(nil)
+                }
+            }
+    }
+    
+    private func fillingParameters (parameters: [SearchParameter]) -> [String : String] {
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/MetAPI.swift:100]: Bad name: `fillingParameters`
This is a method which calculates parameters out of search-parameters. I suggest:
```
func parameters(from searchParameters: [SearchParameter]) -> [String : String]
```

File: TheMET/TheMET/Network/MetAPI.swift [hunk 4]
```diff
@@ -73,5 +73,74 @@ class MetAPI{
                 }
             }
     }
-
+    
+    func search(parameters: [SearchParameter], completion: @escaping (SearchResponse?) -> Void) {
+        var urlString: String = self.urlBaseString
+        let urlStringSuffix: String = "public/collection/v1/search"
+        urlString.append(urlStringSuffix)
+        var searchParameters: [String : String] = self.fillingParameters(parameters: parameters)
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/MetAPI.swift:81]: Wh is is `var`?

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Network/MetAPI.swift (hunk 1 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPI.swift:106]: fixed

File: TheMET/TheMET/Network/MetAPI.swift (hunk 2 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPI.swift:141]: fixed

File: TheMET/TheMET/Network/MetAPI.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPI.swift:100]: fixed

File: TheMET/TheMET/Network/MetAPI.swift (hunk 4 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/MetAPI.swift:81]: fixed

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**